test/scule.test.ts

```diff
@@ -140,6 +147,7 @@ describe("titleCase", () => {
     ["foo", "Foo"],
     ["foo-bar", "Foo Bar"],
     ["this-IS-aTitle", "This is a Title"],
+    ["Foo Bar", "Foo Bar"],
```

Comment:
Maybe add at least to types test as well? (for types we can do simply similar fix with trim should be fine)

Reply:
I made some changes to types and added types test cases for `TrainCase` and `TitleCase`(TrainCase with space joiner). 

There are still two cases didn't pass. That's because `titleCase` and `trainCase` use `filter(Boolean)` to filter empty string. I don't know how to fix the type properly so I add a `@ts-expect-error` comment above it.

Could you do some help?